docs/babel.config.js

```diff
@@ -13,6 +13,7 @@ const alias = {
   '@mui/monorepo': '../node_modules/@mui/monorepo',
   '@mui-internal/docs-utilities': '../node_modules/@mui/monorepo/packages/docs-utilities',
   '@mui/joy': '../node_modules/@mui/monorepo/packages/mui-joy/src',
+  '@mui/base': '../node_modules/@mui/monorepo/packages/mui-base/src',
```

Comment:
This solves this error:
```
error - ../node_modules/@mui/monorepo/packages/mui-joy/src/Checkbox/Checkbox.tsx:6:0
Module not found: Can't resolve '@mui/base/useSwitch'

Import trace for requested module:
../node_modules/@mui/monorepo/packages/mui-joy/src/Checkbox/index.ts
../packages/grid/x-data-grid/src/joy/index.tsx
./data/data-grid/components/JoyUISlots.js
./data/data-grid/components/components.md?@mui/markdown
./pages/x/react-data-grid/components.js
```

I've added the `@mui/joy` alias in https://github.com/mui/mui-x/pull/4854#discussion_r871180005, but now I think we should use the `@mui/joy` and `@mui/base` from the NPM instead.